.github/workflows/poetry-smoke-tests.yaml

```diff
@@ -0,0 +1,33 @@
+name: Poetry Smoke Tests
+
+on:
+  push:
+    branches: [ '**' ] # This will run the action on all branches
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+
+    - name: Install Poetry
+      run: curl -sSL https://install.python-poetry.org | python -
+
+    - name: Install Dependencies
+      run: poetry install
+
+    - name: Build Project
+      run: poetry build
+
+    - name: Start Poetry Shell (Optional)
```

Comment:
Good question - `poetry shell` should create a virtualenv - my thinking was that I wanted to smoke test all the poetry stuff (including poetry install - in case poetry's install script had an issue or they changed their install method, so we'd get an early warning) - but my understanding of the `poetry shell` command is that it creates a virtualenv, activates it and then returns. 

Here's [the docs for `poetry shell`](https://python-poetry.org/docs/cli/#shell) - I'm wondering now if that job should `poetry shell && exit`? Or maybe that exit is not even necessary. 

In local testing at least, the command returns use of the shell to you.

Reply:
You're right - it doesn't work in a GitHub runner due to `ioctl` issues - I don't think it's strictly worth smoke testing, so I removed it.